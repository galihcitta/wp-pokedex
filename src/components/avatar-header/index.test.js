import React from "react";
import AvatarHeader from "./index";
import dummyImage from "../../static/image/dummy-square.jpg";

let wrapper;
let wrap;
let component;

beforeEach(() => {
    component = (
        <AvatarHeader
            name="Burbasur"
            abilities={[
                {
                    ability: {
                        name: "chlorophyll",
                        url: "https://pokeapi.co/api/v2/ability/34/",
                    },
                    is_hidden: true,
                    slot: 3,
                },
                {
                    ability: {
                        name: "overgrow",
                        url: "https://pokeapi.co/api/v2/ability/65/",
                    },
                    is_hidden: false,
                    slot: 1,
                },
            ]}
            image={dummyImage}
        ></AvatarHeader>
    );
    wrapper = shallow(component);
    wrap = mount(component);
});

describe("Header Component", () => {
    it("renders and create snapshot <AvatarHeader />", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it("renders and create snapshot <AvatarHeader.Loading />", () => {
        expect(
            toJson(mount(<AvatarHeader.Loading></AvatarHeader.Loading>)),
        ).toMatchSnapshot();
    });
    it("renders loading <AvatarHeader.Loading />", () => {
        expect(
            mount(<AvatarHeader.Loading></AvatarHeader.Loading>).exists(),
        ).toBe(true);
    });

    it("renders success <AvatarHeader/>", () => {
        expect(wrapper.exists()).toBe(true);
    });
});
