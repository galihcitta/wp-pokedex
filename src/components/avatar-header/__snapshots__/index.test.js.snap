// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders and create snapshot <AvatarHeader /> 1`] = `
<styled.div>
  <div
    className="avatar__header"
  >
    <div
      className="avatar__header__left"
    >
      <div
        className="avatar__header top"
      >
        <div
          className="avatar__header__avatar"
        >
          <img
            alt="Burbasur"
            src="dummy-square.jpg"
          />
        </div>
        <span
          className="avatar__header__title"
        >
          Burbasur
        </span>
      </div>
      <Abilities
        data={
          Array [
            Object {
              "ability": Object {
                "name": "chlorophyll",
                "url": "https://pokeapi.co/api/v2/ability/34/",
              },
              "is_hidden": true,
              "slot": 3,
            },
            Object {
              "ability": Object {
                "name": "overgrow",
                "url": "https://pokeapi.co/api/v2/ability/65/",
              },
              "is_hidden": false,
              "slot": 1,
            },
          ]
        }
      />
    </div>
  </div>
</styled.div>
`;

exports[`Header Component renders and create snapshot <AvatarHeader.Loading /> 1`] = `
.c0 {
  display: block;
}

.c0 .avatar__header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 10px;
  width: 100%;
}

.c0 .avatar__header__left {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0 .avatar__header__title {
  color: #4a4a4a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 24px;
  font-weight: bold;
  padding: 10px;
  text-transform: uppercase;
}

.c0 .avatar__header__title > span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c0 .avatar__header .top {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .avatar__header__avatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #fff;
  border: 1px solid #ddd;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 70px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2px;
  position: relative;
  width: 70px;
}

.c0 .avatar__header__avatar > img {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 50%;
  height: 70px;
  margin: 10px 0;
  width: 70px;
}

.c2 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 0;
  height: 10px;
  margin: 10px 0;
  width: 40%;
}

.c4 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 0;
  height: 10px;
  margin: 10px 0;
  width: 120px;
}

.c5 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 0;
  height: 10px;
  margin: 10px 0;
  width: 40;
}

.c6 {
  -webkit-animation: hnMiWC 1s infinite ease-in-out;
  animation: hnMiWC 1s infinite ease-in-out;
  border-radius: 0;
  height: 10px;
  margin: 10px 0;
  width: 150px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c3 .label-types {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 10px 0;
  width: 100%;
}

.c3 .label-types--box {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Poppins",sans-serif;
  font-size: 12px;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 5px;
  margin-right: 5px;
  padding: 7px 15px;
  padding: 10px;
}

.c3 .label-types--box.is_hidden {
  opacity: 0.5;
}

.c3 .label-types--box > span:last-child {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(214,214,214,0.16);
  border: 1px solid #dcdcdc;
  border-radius: 50%;
  color: #4a4a4a;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 22px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 8px;
  width: 22px;
}

<AvatarHeaderLoading>
  <styled.div>
    <div
      className="c0"
    >
      <div
        className="avatar__header"
      >
        <div
          className="avatar__header__left"
        >
          <div
            className="avatar__header top"
          >
            <div
              className="avatar__header__avatar"
            >
              <Skeleton
                circle={true}
                height={70}
                style={Object {}}
                width="70px"
              >
                <styled.div
                  circle={true}
                  height={70}
                  style={Object {}}
                  width="70px"
                >
                  <div
                    className="c1"
                    height={70}
                    style={Object {}}
                    width="70px"
                  />
                </styled.div>
              </Skeleton>
            </div>
            <span
              className="avatar__header__title"
            >
              <Skeleton
                circle={false}
                height={10}
                style={Object {}}
                width="40%"
              >
                <styled.div
                  circle={false}
                  height={10}
                  style={Object {}}
                  width="40%"
                >
                  <div
                    className="c2"
                    height={10}
                    style={Object {}}
                    width="40%"
                  />
                </styled.div>
              </Skeleton>
            </span>
          </div>
          <AbilitiesLoading>
            <styled.div>
              <div
                className="c3"
              >
                <div
                  className="label-types"
                >
                  <div
                    className="label-types--box is_hidden"
                    test-id="box-attribute"
                  >
                    <span>
                      <Skeleton
                        circle={false}
                        height={10}
                        style={Object {}}
                        width="120px"
                      >
                        <styled.div
                          circle={false}
                          height={10}
                          style={Object {}}
                          width="120px"
                        >
                          <div
                            className="c4"
                            height={10}
                            style={Object {}}
                            width="120px"
                          />
                        </styled.div>
                      </Skeleton>
                    </span>
                    <span>
                      <Skeleton
                        circle={false}
                        height={10}
                        style={Object {}}
                        width="40"
                      >
                        <styled.div
                          circle={false}
                          height={10}
                          style={Object {}}
                          width="40"
                        >
                          <div
                            className="c5"
                            height={10}
                            style={Object {}}
                            width="40"
                          />
                        </styled.div>
                      </Skeleton>
                    </span>
                  </div>
                  <div
                    className="label-types--box"
                    test-id="box-attribute"
                  >
                    <span>
                      <Skeleton
                        circle={false}
                        height={10}
                        style={Object {}}
                        width="150px"
                      >
                        <styled.div
                          circle={false}
                          height={10}
                          style={Object {}}
                          width="150px"
                        >
                          <div
                            className="c6"
                            height={10}
                            style={Object {}}
                            width="150px"
                          />
                        </styled.div>
                      </Skeleton>
                    </span>
                    <span>
                      <Skeleton
                        circle={false}
                        height={10}
                        style={Object {}}
                        width="40"
                      >
                        <styled.div
                          circle={false}
                          height={10}
                          style={Object {}}
                          width="40"
                        >
                          <div
                            className="c5"
                            height={10}
                            style={Object {}}
                            width="40"
                          />
                        </styled.div>
                      </Skeleton>
                    </span>
                  </div>
                </div>
              </div>
            </styled.div>
          </AbilitiesLoading>
        </div>
      </div>
    </div>
  </styled.div>
</AvatarHeaderLoading>
`;
